proposals/extended-property-patterns.md

```diff
@@ -0,0 +1,44 @@
+# Extended property patterns
+
+## Summary
+[summary]: #summary
+
+Allow property subpatterns to reference nested members.
+
+## Motivation
+[motivation]: #motivation
+
+When you want to match a child property, nesting another recursive pattern adds too much noise which will hurt readability with no real advantage.
+
+## Detailed design
+[design]: #detailed-design
+
+The pattern syntax is modified as follow:
+
+```diff
+property_pattern
+  : type? property_pattern_clause simple_designation?
+  ;
+
+property_pattern_clause
+  : '{' (subpattern (',' subpattern)* ','?)? '}'
+  ;
+
+subpattern
+- : identifier ':' pattern
++ : subpattern_name ':' pattern
+  ;
+
++subpattern_name
++ : identifier 
++ | subpattern_name '.' identifier
++ ;
+```
+
+The receiver for each name lookup is the stripped type of the previous member starting from the *input type* of the *property_pattern*.
```

Comment:
>stripped type

This is probably not the correct term to use. How do we refer to such type?

Reply:
Here's the type of language used elsewhere in the spec: `Determine the type T0. If T is a nullable type, T0 is its underlying type, otherwise T0 is equal to T.`

---
In reply to: [601701046](https://github.com/dotnet/csharplang/pull/4585#discussion_r601701046) [](ancestors = 601701046)